src/lib/slice.zig

```diff
@@ -427,23 +427,96 @@ pub const Slice = struct {
         }
     }
 
+    /// Pushes git notes to remote repository to share slice relationships with team
+    pub fn pushNotes(self: Slice, alloc: Allocator) !void {
+        _ = self; // We push all notes, not just for this slice
+        
+        const rr_push_notes = try Git.push(.{
+            .allocator = alloc,
+            .args = &.{
+                // TODO: use proper remote here
+                "origin",
+                "refs/notes/commits:refs/notes/commits",
+            },
+        });
+        defer alloc.free(rr_push_notes.stderr);
+        defer alloc.free(rr_push_notes.stdout);
+        
+        if (rr_push_notes.term.Exited != 0) {
+            log.debug("pushNotes:: push notes stderr: {s}", .{rr_push_notes.stderr});
+            log.err(
+                "pushNotes:: push notes failed with exit code {d}",
+                .{rr_push_notes.term.Exited},
+            );
+            return SparseError.UNABLE_TO_PUSH_SLICE;
+        }
+        
+        log.info("Successfully pushed git notes to remote", .{});
+    }
+
+    /// Fetches git notes from remote repository to get latest slice relationships
+    pub fn fetchNotes(self: Slice, alloc: Allocator) !void {
+        _ = self; // We fetch all notes, not just for this slice
+        
+        const rr_fetch_notes = try Git.fetch(.{
+            .allocator = alloc,
+            .args = &.{
+                // TODO: use proper remote here
+                "origin",
+                "refs/notes/commits:refs/notes/commits",
```

Comment:
either fetch all namespaced notes or the notes only for the self. use notesNamespace of a slice here. I feel like instead of pushing all or pulling all notes it would better to handle that individually for slices.